server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -718,7 +704,7 @@ private ExternalReaderManager createReaderManager(RefreshWarmerListener external
     private DirectoryReader getDirectoryReader() throws IOException {
         // for segment replication: replicas should create the reader from store, we don't want an open IW on replicas.
         if (engineConfig.isReadOnly()) {
-            return DirectoryReader.open(store.directory());
+            return new SoftDeletesDirectoryReaderWrapper(DirectoryReader.open(store.directory()), Lucene.SOFT_DELETES_FIELD);
         }
         return DirectoryReader.open(indexWriter);
```

Comment:
do we not need writeAllDeletes set to true here?  `DirectoryReader.open(indexWriter, true, true);`